Editor/Scripts/Server/EditorServerSettings.cs

```diff
@@ -0,0 +1,51 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+[FilePath("Assets/ProjectSettings/EditorServerSettings.yml", FilePathAttribute.Location.ProjectFolder)]
+internal class EditorServerSettings : ScriptableSingleton<EditorServerSettings> {
+
+    private const string DIRTY_KEY = "MESHSYNC_AUTO_SETUP_SETTINGS_DIRTY";
+    
+    [SerializeField] private ushort m_port = 8081;
+
+    [SerializeField] private bool m_active = true;
+
+    internal ushort Port {
+        get {return m_port; }
+        set {
+            
+            if (m_port == value)
+                return;
+            
+            m_port = value;
+            Save(true);
+
+            if (m_active) {
+                Dirty = true;
+            }
+        }
+    }
+
+    internal bool Active {
```

Comment:
Don't the Active and Dirty fields belong to `EditorServer` instead of `EditorServerSettings` ?

Reply:
Active is a setting, you can have the editor server active or inactive. Dirty refers to any of the settings being dirty

Reply:
> When I read the code, it seems that it's the former

It's mean to be the latter, its a setting if the server should be activated by default. When we change the setting, we also change the server state, so you could in theory use it a state.

> Perhaps it will be clearer if we can separate between active state and setting ?

I agree, will implement this